model/task/task.go

```diff
@@ -739,26 +739,42 @@ func (t *Task) MarkAsUndispatched() error {
 }
 
 // MarkGeneratedTasks marks that the task has generated tasks.
-func MarkGeneratedTasks(taskID string, errorToSet error) error {
-	if adb.ResultsNotFound(errorToSet) || db.IsDuplicateKey(errorToSet) {
+func MarkGeneratedTasks(taskID string) error {
+	query := bson.M{
+		IdKey:             taskID,
+		GeneratedTasksKey: bson.M{"$exists": false},
+	}
+	update := bson.M{
+		"$set": bson.M{
+			GeneratedTasksKey: true,
```

Comment:
If tasks are generated correctly, should we clear GenerateTasksErrorKey? (Is it possible we'd hit these two cases separately?) I realize we didn't do this before either though

Reply:
I'd like to leave this to the caller. Practically both will be cleared already in most scenarios, because usually if you would hit this again for the same task then the task has restarted, which clears these